static/app/components/sidebar/subnav.tsx

```diff
@@ -0,0 +1,63 @@
+import {
+  createContext,
+  Fragment,
+  type RefObject,
+  useContext,
+  useLayoutEffect,
+  useRef,
+} from 'react';
+import {createPortal} from 'react-dom';
+import styled from '@emotion/styled';
+
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+
+interface SubnavContextData {
+  containerRef: RefObject<HTMLDivElement>;
+}
+
+const subnavContext = createContext<SubnavContextData>({
+  containerRef: {current: null},
+});
+
+export function SubnavContainer({children}) {
+  const containerRef = useRef<HTMLDivElement>(null);
+  useLayoutEffect(() => {
+    if (!containerRef.current) return;
+    if (containerRef.current.children.length === 0) {
+      document.body.dataset.navLevel = '1';
+    } else {
+      document.body.dataset.navLevel = '2';
+    }
+  });
```

Comment:
Not the cleanest pattern, but since the sidebar uses `position: fixed`, we need an effect that surfaces how many navigation levels are active so we can adjust the static app layout from inside the render tree.

Reply:
heh
would there ever be a world where we have more layers than just 2?

Reply:
Good question! I don't imagine there would be, so this could be treated as a `boolean` attribute or class name.